File: frontend/src/types/proposal.ts
export type ProposalSource = "AI_DRAFT" | "HUMAN";
export type ProposalStatus = "DRAFT" | "APPROVED" | "SENT" | "REJECTED";

export interface Proposal {
  id: number;
  advertiser_id: number;
  subject: string;
  body: string;
  source: ProposalSource;
  status: ProposalStatus;
  needs_human: boolean;
  needs_human_reason: string | null;
  model_used: string | null;
  created_by_id: number | null;
  approved_by_id: number | null;
  created_at: string;
  updated_at: string;
  approved_at: string | null;
  sent_at: string | null;
}

File: backend/app/schemas/proposal.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.proposal import ProposalSource, ProposalStatus


class ProposalOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    advertiser_id: int
    subject: str
    body: str
    source: ProposalSource
    status: ProposalStatus
    needs_human: bool
    needs_human_reason: str | None
    model_used: str | None
    created_by_id: int | None
    approved_by_id: int | None
    created_at: datetime
    updated_at: datetime
    approved_at: datetime | None
    sent_at: datetime | None


class ProposalCreate(BaseModel):
    subject: str = Field(min_length=1, max_length=200)
    body: str = Field(min_length=1)


class ProposalUpdate(BaseModel):
    subject: str | None = Field(default=None, min_length=1, max_length=200)
    body: str | None = Field(default=None, min_length=1)
